### ISSUE TITLE
contacts/iphone/new record

### ISSUE THREAD

**Issue by mariannoemsk** [owncloud/contacts#770]
I am struggling to find out why, iPhone is unable to add new contact to own cloud (CarDav).
When I add new contact in iPhone, it should automatically add new contact to own cloud as well. However once I add the contact it starts to upload details (small circle at top) and than contact disappear. No record in own cloud at all. No log error found as well. Any idea what that cause this? 
System:debian 7
MySQL
Apache 2
iPhone IOS7

thanks

**Comment by oracle10gr2** [comment]
Yes, definitly, that is because new contact id  looks  like 'XXX-XXX-XXX-XXX@HHH', uploaded into IOS  then the request  for update will be url encoded twice on the  IOS side ->>...XX-XXX%40HHH  -->>  ..XXX-XXX%2540HHH and  then  received by OC7  but decoded  once -->>XX-XXX%40HHH  so we get wrong contact UID and so it disappeared or changes lost....
The best solution IMHO, don't use '@' sign as delimiter at all, using '-' instead

The solution described here:
https://github.com/owncloud/contacts/issues/772

Max
